snapcraft/plugins/nodejs.py

```diff
@@ -72,8 +74,12 @@ def schema(cls):
 
     def __init__(self, name, options):
         super().__init__(name, options)
-        self._nodejs_tar = sources.Tar(_get_nodejs_release(),
-                                       os.path.join(self.partdir, 'npm'))
+        try:
+            self._nodejs_tar = sources.Tar(
+                    _get_nodejs_release(self.options.node_engine),
+                    os.path.join(self.partdir, 'npm'))
+        except AttributeError:
+            raise AttributeError('node-engine not defined in snapcraft.yaml')
```

Comment:
Not _you_, but the plugin currently deletes it on line 71. If you add the `node-engine` option to the `required` key, snapcraft will catch it when validating the YAML instead of your needing to do it here. See the copy plugin for an example of what you'll probably need to do.